### PR TITLE
fix bug in autograd type() for non-default GPU input

### PR DESCRIPTION
**Pull request by soumith**
Simple failure case where gradients of xv would be on device 0 in master, instead of device 1.

```python
import torch
from torch.autograd import Variable

x = torch.randn(10).cuda(1)
xv = Variable(x, requires_grad=True)
y = xv.cpu()
y.sum().backward()

print(xv.grad.get_device() == 1)
```

### CODE REVIEW

**Comment by soumith** [comment]
cc: @colesbury for your upcoming PR

**Review by colesbury — APPROVED** [review]
lgtm

**Comment by apaszke** [comment]
Whaat? `type()` should work with auto-GPU right?

**Comment by soumith** [comment]
it is Variable type. it does work with auto-GPU, but the device ID is lost for the backward. It puts it on the current GPU.
